Formula/harmony-one.rb

```diff
@@ -0,0 +1,113 @@
+  depends_on macos: :high_sierra
+  depends_on "make"
+  depends_on "openssl@1.1"
+  
+
+  # CONTRIBUTORS: Keep resources updated according to the changelog release tag:
+  #     - https://github.com/harmony-one/harmony/releases/tag/v#{version}
+
+  resource "bls" do
+    url "https://github.com/harmony-one/bls.git",
+        revision: "2b7e49894c0f15f5c40cf74046505b7f74946e52"
+  end
+
+  resource "mcl" do
+    url "https://github.com/harmony-one/mcl.git",
+        revision: "99e9aa76e84415e753956c618cbc662b2f373df1"
+  end
+
+  def install
+    # See https://github.com/golang/go/issues/26487
+    ENV.O0 if OS.linux?
+
+    # Setup Build Paths
+    ENV["GOPATH"] = buildpath
+
+    harmony_path = buildpath/"src/github.com/harmony-one/harmony"
+    harmony_path.install buildpath.children
+
+    %w[mcl bls].each do |r|
+      temp_dir = buildpath/"src/github.com/harmony-one/#{r}"
+
+      resource(r).stage do
+        mv pwd, temp_dir
+      end
+    end
+
+    cd buildpath/"src/github.com/harmony-one/harmony"
+
+    # Correct Directory Locations In Build Scripts
+    # TODO: Switch to brew 'inreplace' formula method
+    system "sed -i -e 's/openssl/openssl\@1\.1/g' Makefile"
+    system "sed -i -e 's/openssl/openssl\@1\.1/g' scripts/go_executable_build.sh"
+
+    # Build Binarys
+    system "go", "mod", "tidy"
+    system "make "
+
+    bin.install "./bin/harmony" => "harmony"
+    bin.install "./bin/bootnode" => "bootnode"
+    lib.install Dir["./bin/*.{dylib,so}"]
+  end
+
+  def caveats
+    <<~EOS
+      # Harmony-One Blockchain Node
+      "Run a Harmony One blockchain Node"
+
+      NOTICE: Due to Macos Security you may need to allow
+      the application under Settings > Security & Privacy > General
+      You may also need to allow network access depending on your firewall
+      configuration.
+
+      USAGE: harmony
+
+      If you do not have any keys or ONE wallet configured you will#{"  "}
+      get the following error:
+
+          ERROR when loading bls key: stat ./.hmy/blskeys: no such file or directory
+
+      You can create these with the following commands:
+          # Download the GO SDK
+          curl -O https://raw.githubusercontent.com/harmony-one/go-sdk/master/scripts/hmy.sh
+      #{"    "}
+          # Make the wrapper script for the binaries exacutable#{"   "}
+          chmod u+x hmy.sh
+
+          # Create wallet
+          ./hmy.sh -d
+          ./hmy keys add test-account --passphrase
+
+      Find more documentation at:
+
+      https://github.com/harmony-one/harmony
+      https://docs.harmony.one/home/network/wallets/harmony-cli/download-setup#1-for-linux-pure-statically-linked-binary
+      https://docs.harmony.one/home/network/wallets/harmony-cli/create-import-wallet
+
+    EOS
+  end
+
+  test do
+    # Basic version check
+    assert true, `bootnode --version`
+    assert true, `harmony --version`
+
+    # Test that the binary fails with bls key error
+    refute_match "/ERROR(.*)(bls.*)/", `harmony`
```

Comment:
```suggestion
    assert_match /ERROR(.*)(bls.*)/, shell_output("harmony")
```

Reply:
It fails when you use assert.

Reply:
its an error so you have to handle it. Assert looks for true and this creates a false value already tried this

Reply:
Ok i was wrong here must have used bad regex the other times -__-